tests/js/server/aql/aql-index-hints.js

```diff
@@ -89,6 +85,61 @@ function ahuacatlSkiplistOverlappingTestSuite () {
       assertEqual(usedIndexes[0].length, 1);
       assertEqual(usedIndexes[0][0], defaultEqualityIndex);
     },
+    
+    testPrimary : function () {
+      [ 
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: false} FILTER doc._key == 'test' RETURN doc`,
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: true} FILTER doc._key == 'test' RETURN doc`,
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: false} FILTER doc._key == 'test' && doc.testi == 99 RETURN doc`,
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: true} FILTER doc._key == 'test' && doc.testi == 99 RETURN doc`,
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: false} FILTER doc.testi == 99 && doc._key == 'test' RETURN doc`,
+        `FOR doc IN ${cn} OPTIONS {indexHint: 'primary', forceIndexHint: true} FILTER doc.testi == 99 && doc._key == 'test' RETURN doc`,
```

Comment:
Shouldn't we use `doc.a` instead of `doc.testi`? As far as I can see there is no index on `testi`, so there is no reason for the optimizer to pick an other index than the primary, or am I missing something?

Reply:
The idea of this test is to check if index hints cause problems if the hint can only be used for a part of the filter condition.

For example, the filter condition here is `doc.testi == 99 && doc._key == 'test'`. The index hint is `primary`. Only the `doc._key == 'test'` subcondition can make use of the index hint here. The other subcondition `doc.testi == 99` cannot (not only because there is no index on it, but also because the index hint is "primary").

That was basically the whole point of the issue: what happens if there are multiple subconditions, of which only some can make use of a forced index hint. The case in question was using OR-combined conditions though, e.g. `doc.testi == 99 || doc._key == 'test'`.

Reply:
Got it.